nemo/collections/tts/modules/common.py

```diff
@@ -82,35 +82,36 @@ def __init__(self, input_size, hidden_size, num_layers=1, lstm_norm_fn="spectral
 
         self.bilstm.flatten_parameters()
 
-    def lstm_tensor(self, context: Tensor, lens: Tensor, enforce_sorted: bool = False) -> Tuple[Tensor, Tensor]:
-        seq = nn.utils.rnn.pack_padded_sequence(
-            context, lens.long().cpu(), batch_first=True, enforce_sorted=enforce_sorted
-        )
-        return self.lstm_sequence(seq)
+    def lstm_sorted(self, context: Tensor, lens: Tensor, hx: Optional[Tuple[Tensor, Tensor]] = None) -> Tensor:
+        seq = nn.utils.rnn.pack_padded_sequence(context, lens.long().cpu(), batch_first=True, enforce_sorted=True)
+        ret, _ = self.bilstm(seq, hx)
+        return nn.utils.rnn.pad_packed_sequence(ret, batch_first=True)[0]
 
-    def lstm_sequence(self, seq: PackedSequence) -> Tuple[Tensor, Tensor]:
-        if not (torch.jit.is_scripting() or torch.jit.is_tracing()):
-            self.bilstm.flatten_parameters()
-            ret, _ = self.bilstm(seq)
-        else:
-            # Calculate sizes and prepare views to our zero buffer to pass as hx
-            max_batch_size = seq.batch_sizes[0]
-            common_shape = (self.bilstm.num_layers * 2, max_batch_size)
-            hx = (
-                seq.data.new_zeros(*common_shape, self.real_hidden_size),
-                seq.data.new_zeros(*common_shape, self.bilstm.hidden_size),
-            )
-            ret, _ = self.bilstm(seq, hx)
-        return nn.utils.rnn.pad_packed_sequence(ret, batch_first=True)
-
-    def forward(self, context: Tensor, lens: Tensor) -> Tensor:
+    def lstm(self, context: Tensor, lens: Tensor, hx: Optional[Tuple[Tensor, Tensor]] = None) -> Tensor:
+        # To be ONNX-exportable, we need to sort here rather that while packing
         context, lens, unsort_ids = sort_tensor(context, lens)
+        # Calculate sizes and prepare views to our zero buffer to pass as hx
+        max_batch_size = context.shape[0]
+        common_shape = (self.bilstm.num_layers * 2, max_batch_size)
+        hx = (
+            context.new_zeros(*common_shape, self.real_hidden_size),
+            context.new_zeros(*common_shape, self.bilstm.hidden_size),
+        )
+        ret = self.lstm_sorted(context, lens, hx=hx)
+        return ret[unsort_ids]
+
+    def lstm_nocast(self, context: Tensor, lens: Tensor) -> Tensor:
         dtype = context.dtype
-        # this is only needed for Torchscript to run in Triton
+        # autocast guard is only needed for Torchscript to run in Triton
         # (https://github.com/pytorch/pytorch/issues/89241)
         with torch.cuda.amp.autocast(enabled=False):
-            ret = self.lstm_tensor(context.to(dtype=torch.float32), lens, enforce_sorted=True)
-        return ret[0].to(dtype=dtype)[unsort_ids]
+            return self.lstm(context.to(dtype=torch.float32), lens).to(dtype=dtype)
+
+    def forward(self, context: Tensor, lens: Tensor) -> Tensor:
+        self.bilstm.flatten_parameters()
```

Comment:
Do we need this? It's called in init

Reply:
Well that may be required for some distributed envirionments. It amounts to almost no-op when weights are already flat.